math/src/field/smallprimefield/generic_prime_field_elt.rs

```diff
@@ -0,0 +1,426 @@
+use crate::field::{
+    traits::{FieldElement, StarkField},
+    QuadExtensionA,
+};
+use core::{
+    convert::{TryFrom, TryInto},
+    fmt::{Debug, Display, Formatter},
+    mem,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+    slice,
+};
+use utils::{
+    collections::Vec,
+    string::{String, ToString},
+    AsBytes, ByteReader, ByteWriter, Deserializable, DeserializationError, Randomizable,
+    Serializable,
+};
+
+// Number of bytes needed to represent field element
+const ELEMENT_BYTES: usize = mem::size_of::<u64>();
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Default)]
+pub struct GenericPrimeFieldElt<const M: u64, const G: u64> {
+    pub(crate) value: u64,
+}
+
+impl<const M: u64, const G: u64> GenericPrimeFieldElt<M, G> {
+    // TODO: this may need to be something else, e.g. its own type param
+    const fn get_twoadic_root() -> Self {
+        Self::new(G)
+    }
```

Comment:
Is this function needed? I think it can be replaced in favor of using `Self::new(G)` directly.

Reply:
This wasn't actually the right value. Updated now.